Add warning not to rely on exact vector metric value (#1688)
hazelcast/hz-docs: docs/modules/data-structures/pages/vector-collections.adoc

TIP: The recommended method for computing cosine similarity is to normalize all vectors to unit length and use the DOT metric instead.
The COSINE metric can be significantly slower than the DOT metric, especially when the xref:jvm-configuration[Vector API] is not enabled.

> WARNING: You should not rely on exact value of the metric. 

Reviewer: ```suggestion
WARNING: Do not rely on the exact value of the metric. 
```

> It is a floating point number and the exact value may depend on the order of operations and rounding during evaluation which can be different in different implementations (with or without vectorization) and in different Hazelcast versions.

Reviewer: ```suggestion
It is a floating point number and the exact value may depend on the order of operations and rounding during evaluation which can vary in different implementations (with or without vectorization) and in different Hazelcast versions.
```
